client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/RemoteShuffleInputGateDelegation.java

```diff
@@ -260,8 +261,10 @@ private Consumer<Throwable> getFailureListener(ResultPartitionID rpID) {
         if (cause != null) {
           return;
         }
-        Class<?> clazz = PartitionUnRetryAbleException.class;
-        if (throwable.getMessage() != null && throwable.getMessage().contains(clazz.getName())) {
+        String message = throwable.getMessage();
+        if (message != null
+            && (message.contains(CelebornIOException.class.getName())
```

Comment:
`openStreamInternal` could trigger an exception, but it should not be considered a data consumption error because the task should retry for it. Maybe this change is too generic.

Reply:
@FMX, is it better to change some `CelebornIOException` regarded as data consumption error to `PartitionUnRetryAbleException`?

Reply:
Celeborn worker throw a PartitionUnRetryAbleException when it determines that a file is unreadable or does not exist; otherwise, the client should retry for CelebornIOException. So if we can clearly identify these cases, we can directly convert them to PartitionUnRetryAbleException. (In addition we already have `PartitionConnectionException` for client-worker connection problem.)